lib/replicator.js

```diff
@@ -1674,12 +1674,13 @@ function clampRange (core, r) {
     const start = core.bitfield.firstUnset(r.start)
 
     if (r.end === -1) r.start = start === -1 ? core.tree.length : start
-    else if (start === -1) r.start = r.end
+    else if (start === -1 || start >= r.end) r.start = r.end
     else {
-      const end = core.bitfield.lastUnset(r.end)
-
       r.start = start
-      r.end = end
+
+      const end = core.bitfield.lastUnset(r.end - 1)
+
+      if (end !== -1) r.end = end + 1
```

Comment:
This now correctly returns `-1` when no last unset bit is found so we need to guard against that.